pkg/volume/iscsi/attacher.go

```diff
@@ -0,0 +1,205 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package iscsi
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/golang/glog"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/kubernetes/pkg/util/mount"
+	"k8s.io/kubernetes/pkg/volume"
+	volumeutil "k8s.io/kubernetes/pkg/volume/util"
+	"k8s.io/utils/exec"
+)
+
+type iscsiAttacher struct {
+	host    volume.VolumeHost
+	manager diskManager
+	exe     exec.Interface
+}
+
+var _ volume.Attacher = &iscsiAttacher{}
+
+var _ volume.AttachableVolumePlugin = &iscsiPlugin{}
+
+func (plugin *iscsiPlugin) NewAttacher() (volume.Attacher, error) {
+	return &iscsiAttacher{
+		host:    plugin.host,
+		manager: &ISCSIUtil{},
+		exe:     exec.New(),
+	}, nil
+}
+
+func (plugin *iscsiPlugin) GetDeviceMountRefs(deviceMountPath string) ([]string, error) {
+	mounter := plugin.host.GetMounter()
+	return mount.GetMountRefs(mounter, deviceMountPath)
+}
+
+func (attacher *iscsiAttacher) Attach(spec *volume.Spec, nodeName types.NodeName) (string, error) {
+	return "", nil
+}
+
+func (attacher *iscsiAttacher) VolumesAreAttached(specs []*volume.Spec, nodeName types.NodeName) (map[*volume.Spec]bool, error) {
+	volumesAttachedCheck := make(map[*volume.Spec]bool)
+	for _, spec := range specs {
+		volumesAttachedCheck[spec] = true
```

Comment:
@jingxu97 
Since iSCSI plugin doesn't have a way to confirm whether the volume is attached or not from outside of the kubelet node, we think this method needs to return true always. This is similar to the FC plugin attacher/detacher implementation.